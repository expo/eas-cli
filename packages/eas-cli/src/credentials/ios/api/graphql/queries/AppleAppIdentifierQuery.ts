import { print } from 'graphql';
import gql from 'graphql-tag';

import { ExpoGraphqlClient } from '../../../../../commandUtils/context/contextUtils/createGraphqlClient';
import { withErrorHandlingAsync } from '../../../../../graphql/client';
import {
  AppleAppIdentifierByBundleIdQuery,
  AppleAppIdentifierFragment,
} from '../../../../../graphql/generated';
import { AppleAppIdentifierFragmentNode } from '../../../../../graphql/types/credentials/AppleAppIdentifier';

export const AppleAppIdentifierQuery = {
  async byBundleIdentifierAsync(
    graphqlClient: ExpoGraphqlClient,
    accountName: string,
    bundleIdentifier: string
  ): Promise<AppleAppIdentifierFragment | null> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleAppIdentifierByBundleIdQuery>(
          gql`
            query AppleAppIdentifierByBundleIdQuery(
              $accountName: String!
              $bundleIdentifier: String!
            ) {
              account {
                byName(accountName: $accountName) {
                  id
                  appleAppIdentifiers(bundleIdentifier: $bundleIdentifier) {
                    id
                    ...AppleAppIdentifierFragment
                  }
                }
              }
            }
            ${print(AppleAppIdentifierFragmentNode)}
          `,
          {
            accountName,
            bundleIdentifier,
          },
          {
            additionalTypenames: ['AppleAppIdentifier'],
          }
        )
        .toPromise()
    );
    return data.account.byName.appleAppIdentifiers[0];
  },
};
