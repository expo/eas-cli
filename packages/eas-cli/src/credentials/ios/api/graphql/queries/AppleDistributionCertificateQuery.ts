import assert from 'assert';
import { print } from 'graphql';
import gql from 'graphql-tag';

import { graphqlClient, withErrorHandlingAsync } from '../../../../../graphql/client';
import {
  AppleDistributionCertificateByAccountQuery,
  AppleDistributionCertificateByAppQuery,
  AppleDistributionCertificateFragment,
  IosDistributionType,
} from '../../../../../graphql/generated';
import { AppleDistributionCertificateFragmentNode } from '../../../../../graphql/types/credentials/AppleDistributionCertificate';
import { AppleTeamFragmentNode } from '../../../../../graphql/types/credentials/AppleTeam';

const AppleDistributionCertificateQuery = {
  async getForAppAsync(
    projectFullName: string,
    {
      appleAppIdentifierId,
      iosDistributionType,
    }: { appleAppIdentifierId: string; iosDistributionType: IosDistributionType }
  ): Promise<AppleDistributionCertificateFragment | null> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleDistributionCertificateByAppQuery>(
          gql`
            query AppleDistributionCertificateByAppQuery(
              $projectFullName: String!
              $appleAppIdentifierId: String!
              $iosDistributionType: IosDistributionType!
            ) {
              app {
                byFullName(fullName: $projectFullName) {
                  id
                  iosAppCredentials(filter: { appleAppIdentifierId: $appleAppIdentifierId }) {
                    id
                    iosAppBuildCredentialsArray(
                      filter: { iosDistributionType: $iosDistributionType }
                    ) {
                      id
                      distributionCertificate {
                        id
                        ...AppleDistributionCertificateFragment
                        appleTeam {
                          id
                          ...AppleTeamFragment
                        }
                      }
                    }
                  }
                }
              }
            }
            ${print(AppleDistributionCertificateFragmentNode)}
            ${print(AppleTeamFragmentNode)}
          `,
          {
            projectFullName,
            appleAppIdentifierId,
            iosDistributionType,
          },
          {
            additionalTypenames: ['IosAppCredentials', 'IosAppBuildCredentials'],
          }
        )
        .toPromise()
    );
    assert(data.app, 'GraphQL: `app` not defined in server response');
    return (
      data.app.byFullName.iosAppCredentials[0]?.iosAppBuildCredentialsArray[0]
        ?.distributionCertificate ?? null
    );
  },
  async getAllForAccount(accountName: string): Promise<AppleDistributionCertificateFragment[]> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleDistributionCertificateByAccountQuery>(
          gql`
            query AppleDistributionCertificateByAccountQuery($accountName: String!) {
              account {
                byName(accountName: $accountName) {
                  id
                  appleDistributionCertificates {
                    id
                    ...AppleDistributionCertificateFragment
                  }
                }
              }
            }
            ${print(AppleDistributionCertificateFragmentNode)}
          `,
          {
            accountName,
          }
        )
        .toPromise()
    );
    return data.account.byName.appleDistributionCertificates;
  },
};

export { AppleDistributionCertificateQuery };
