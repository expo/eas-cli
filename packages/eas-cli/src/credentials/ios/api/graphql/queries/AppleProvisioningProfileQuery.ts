import assert from 'assert';
import { print } from 'graphql';
import gql from 'graphql-tag';

import { ExpoGraphqlClient } from '../../../../../commandUtils/context/contextUtils/createGraphqlClient';
import { withErrorHandlingAsync } from '../../../../../graphql/client';
import {
  AppleAppIdentifierFragment,
  AppleDeviceFragment,
  AppleProvisioningProfileFragment,
  AppleProvisioningProfilesByAppQuery,
  AppleTeamFragment,
  IosDistributionType,
} from '../../../../../graphql/generated';
import { AppleAppIdentifierFragmentNode } from '../../../../../graphql/types/credentials/AppleAppIdentifier';
import { AppleDeviceFragmentNode } from '../../../../../graphql/types/credentials/AppleDevice';
import { AppleProvisioningProfileFragmentNode } from '../../../../../graphql/types/credentials/AppleProvisioningProfile';
import { AppleTeamFragmentNode } from '../../../../../graphql/types/credentials/AppleTeam';

export type AppleProvisioningProfileQueryResult = AppleProvisioningProfileFragment & {
  appleTeam?: AppleTeamFragment | null;
} & { appleDevices: AppleDeviceFragment[] } & { appleAppIdentifier: AppleAppIdentifierFragment };
export const AppleProvisioningProfileQuery = {
  async getForAppAsync(
    graphqlClient: ExpoGraphqlClient,
    projectFullName: string,
    {
      appleAppIdentifierId,
      iosDistributionType,
    }: { appleAppIdentifierId: string; iosDistributionType: IosDistributionType }
  ): Promise<AppleProvisioningProfileQueryResult | null> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleProvisioningProfilesByAppQuery>(
          gql`
            query AppleProvisioningProfilesByAppQuery(
              $projectFullName: String!
              $appleAppIdentifierId: String!
              $iosDistributionType: IosDistributionType!
            ) {
              app {
                byFullName(fullName: $projectFullName) {
                  id
                  iosAppCredentials(filter: { appleAppIdentifierId: $appleAppIdentifierId }) {
                    id
                    iosAppBuildCredentialsList(
                      filter: { iosDistributionType: $iosDistributionType }
                    ) {
                      id
                      provisioningProfile {
                        id
                        ...AppleProvisioningProfileFragment
                        appleTeam {
                          id
                          ...AppleTeamFragment
                        }
                        appleDevices {
                          id
                          ...AppleDeviceFragment
                        }
                        appleAppIdentifier {
                          id
                          ...AppleAppIdentifierFragment
                        }
                      }
                    }
                  }
                }
              }
            }
            ${print(AppleProvisioningProfileFragmentNode)}
            ${print(AppleTeamFragmentNode)}
            ${print(AppleDeviceFragmentNode)}
            ${print(AppleAppIdentifierFragmentNode)}
          `,
          {
            projectFullName,
            appleAppIdentifierId,
            iosDistributionType,
          },
          {
            additionalTypenames: [
              'AppleProvisioningProfile',
              'IosAppCredentials',
              'IosAppBuildCredentials',
            ],
          }
        )
        .toPromise()
    );
    assert(data.app, 'GraphQL: `app` not defined in server response');
    return (
      data.app.byFullName.iosAppCredentials[0]?.iosAppBuildCredentialsList[0]
        ?.provisioningProfile ?? null
    );
  },
};
