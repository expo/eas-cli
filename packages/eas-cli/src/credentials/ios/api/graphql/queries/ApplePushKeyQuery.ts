import { print } from 'graphql';
import gql from 'graphql-tag';

import { ExpoGraphqlClient } from '../../../../../commandUtils/context/contextUtils/createGraphqlClient';
import { withErrorHandlingAsync } from '../../../../../graphql/client';
import { ApplePushKeyByAccountQuery, ApplePushKeyFragment } from '../../../../../graphql/generated';
import { ApplePushKeyFragmentNode } from '../../../../../graphql/types/credentials/ApplePushKey';

export const ApplePushKeyQuery = {
  async getAllForAccountAsync(
    graphqlClient: ExpoGraphqlClient,
    accountName: string
  ): Promise<ApplePushKeyFragment[]> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<ApplePushKeyByAccountQuery>(
          gql`
            query ApplePushKeyByAccountQuery($accountName: String!) {
              account {
                byName(accountName: $accountName) {
                  id
                  applePushKeys {
                    id
                    ...ApplePushKeyFragment
                  }
                }
              }
            }
            ${print(ApplePushKeyFragmentNode)}
          `,
          {
            accountName,
          },
          {
            additionalTypenames: ['ApplePushKey'],
          }
        )
        .toPromise()
    );
    return data.account.byName.applePushKeys;
  },
};
