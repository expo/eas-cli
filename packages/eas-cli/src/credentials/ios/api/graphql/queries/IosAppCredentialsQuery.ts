import assert from 'assert';
import { print } from 'graphql';
import gql from 'graphql-tag';

import { ExpoGraphqlClient } from '../../../../../commandUtils/context/contextUtils/createGraphqlClient';
import { withErrorHandlingAsync } from '../../../../../graphql/client';
import {
  CommonIosAppCredentialsFragment,
  CommonIosAppCredentialsWithBuildCredentialsByAppIdentifierIdQuery,
  IosAppCredentialsWithBuildCredentialsByAppIdentifierIdQuery,
  IosDistributionType,
} from '../../../../../graphql/generated';
import { IosAppBuildCredentialsFragmentNode } from '../../../../../graphql/types/credentials/IosAppBuildCredentials';
import {
  CommonIosAppCredentialsFragmentNode,
  CommonIosAppCredentialsWithoutBuildCredentialsFragmentNode,
} from '../../../../../graphql/types/credentials/IosAppCredentials';

export const IosAppCredentialsQuery = {
  async withBuildCredentialsByAppIdentifierIdAsync(
    graphqlClient: ExpoGraphqlClient,
    projectFullName: string,
    {
      appleAppIdentifierId,
      iosDistributionType,
    }: {
      appleAppIdentifierId: string;
      iosDistributionType?: IosDistributionType;
    }
  ): Promise<CommonIosAppCredentialsFragment | null> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<IosAppCredentialsWithBuildCredentialsByAppIdentifierIdQuery>(
          gql`
            query IosAppCredentialsWithBuildCredentialsByAppIdentifierIdQuery(
              $projectFullName: String!
              $appleAppIdentifierId: String!
              $iosDistributionType: IosDistributionType
            ) {
              app {
                byFullName(fullName: $projectFullName) {
                  id
                  iosAppCredentials(filter: { appleAppIdentifierId: $appleAppIdentifierId }) {
                    id
                    ...CommonIosAppCredentialsWithoutBuildCredentialsFragment
                    iosAppBuildCredentialsList(
                      filter: { iosDistributionType: $iosDistributionType }
                    ) {
                      id
                      ...IosAppBuildCredentialsFragment
                    }
                  }
                }
              }
            }
            ${print(CommonIosAppCredentialsWithoutBuildCredentialsFragmentNode)}
            ${print(IosAppBuildCredentialsFragmentNode)}
          `,
          {
            projectFullName,
            appleAppIdentifierId,
            iosDistributionType,
          },
          {
            additionalTypenames: ['IosAppCredentials', 'IosAppBuildCredentials'],
          }
        )
        .toPromise()
    );
    assert(data.app, 'GraphQL: `app` not defined in server response');
    return data.app.byFullName.iosAppCredentials[0] ?? null;
  },
  async withCommonFieldsByAppIdentifierIdAsync(
    graphqlClient: ExpoGraphqlClient,
    projectFullName: string,
    {
      appleAppIdentifierId,
    }: {
      appleAppIdentifierId: string;
    }
  ): Promise<CommonIosAppCredentialsFragment | null> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<CommonIosAppCredentialsWithBuildCredentialsByAppIdentifierIdQuery>(
          gql`
            query CommonIosAppCredentialsWithBuildCredentialsByAppIdentifierIdQuery(
              $projectFullName: String!
              $appleAppIdentifierId: String!
            ) {
              app {
                byFullName(fullName: $projectFullName) {
                  id
                  iosAppCredentials(filter: { appleAppIdentifierId: $appleAppIdentifierId }) {
                    id
                    ...CommonIosAppCredentialsFragment
                  }
                }
              }
            }
            ${CommonIosAppCredentialsFragmentNode}
          `,
          {
            projectFullName,
            appleAppIdentifierId,
          },
          {
            additionalTypenames: ['IosAppCredentials', 'IosAppBuildCredentials', 'App'],
          }
        )
        .toPromise()
    );
    assert(data.app, 'GraphQL: `app` not defined in server response');
    return data.app.byFullName.iosAppCredentials[0] ?? null;
  },
};
