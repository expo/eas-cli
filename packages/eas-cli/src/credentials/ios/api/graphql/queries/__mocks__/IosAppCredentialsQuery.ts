export const IosAppCredentialsQuery = {
  withCommonFieldsByAppIdentifierIdAsync: jest.fn().mockImplementation(() => {
    return {
      id: '48d0cd34-96d8-447c-9de8-40dd2c3ca0f8',
      app: {
        id: '033134f3-e85c-4bde-9065-41a9c066ec31',
        fullName: '@quinlanj/test52',
        __typename: 'App',
      },
      appleTeam: {
        id: '6d39dbfe-a30a-44a4-bc36-d562e5963fa3',
        appleTeamIdentifier: 'test-apple-identifier',
        appleTeamName: null,
        __typename: 'AppleTeam',
      },
      appleAppIdentifier: {
        id: 'bd9c3609-26b9-451e-8ba6-e0604736c7fc',
        bundleIdentifier: 'com.quinlanj.test52',
        __typename: 'AppleAppIdentifier',
      },
      pushKey: {
        id: '74bdd007-da7b-4190-a571-edb87e9324a1',
        keyIdentifier: '9839M6AY8W',
        updatedAt: '2020-10-13T18:39:04.463Z',
        appleTeam: {
          id: '6d39dbfe-a30a-44a4-bc36-d562e5963fa3',
          appleTeamIdentifier: '77KQ969CHE',
          appleTeamName: null,
          __typename: 'AppleTeam',
        },
        __typename: 'ApplePushKey',
      },
      appStoreConnectApiKeyForSubmissions: {
        id: 'b8fe2225-3024-4ebe-9e36-7b9022d3b13e',
        appleTeam: {
          id: 'ca27f9f6-c7cb-40d1-8c41-c168e8548261',
          appleTeamIdentifier: '77KQ969CHE',
          appleTeamName: null,
          __typename: 'AppleTeam',
        },
        issuerIdentifier: '69a6de93-e4e4-47e3-e053-5b8c7c11a4d1',
        keyIdentifier: 'NYN468XNN3',
        name: 'Expo Submissions pl94k8_lnS',
        roles: ['ADMIN'],
        updatedAt: '2020-10-13T18:39:04.463Z',
        __typename: 'AppStoreConnectApiKey',
      },
      iosAppBuildCredentialsList: [
        {
          id: '6f8d0175-0caf-4922-b09b-eae757b83144',
          iosDistributionType: 'APP_STORE',
          distributionCertificate: {
            id: '8b3baa53-b149-4a1b-842b-75ff6e0bcb1e',
            certificateP12: 'test-cert-p12',
            certificatePassword: 'test-cert-password',
            serialNumber: 'test-serial',
            developerPortalIdentifier: 'test-portal-id',
            validityNotBefore: '2020-10-13T18:28:44.000Z',
            validityNotAfter: '2021-10-13T18:28:44.000Z',
            updatedAt: '2020-10-13T18:38:45.275Z',
            appleTeam: {
              id: '6d39dbfe-a30a-44a4-bc36-d562e5963fa3',
              appleTeamIdentifier: 'test-apple-identifier',
              appleTeamName: null,
              __typename: 'AppleTeam',
            },
            __typename: 'AppleDistributionCertificate',
          },
          provisioningProfile: {
            id: '01517f14-b4cf-489e-b1fd-3f5a61cb8df8',
            expiration: '2021-10-13T18:28:44.000Z',
            developerPortalIdentifier: 'test-profile-id',
            provisioningProfile: 'test-profile',
            updatedAt: '2020-10-29T00:18:32.130Z',
            status: 'active',
            appleDevices: [],
            appleTeam: {
              id: '6d39dbfe-a30a-44a4-bc36-d562e5963fa3',
              appleTeamIdentifier: 'test-apple-identifier',
              appleTeamName: null,
              __typename: 'AppleTeam',
            },
            __typename: 'AppleProvisioningProfile',
          },
          __typename: 'IosAppBuildCredentials',
        },
        {
          id: '3e1907a8-7114-4f14-9964-73467bc010ea',
          iosDistributionType: 'AD_HOC',
          distributionCertificate: {
            id: '8b3baa53-b149-4a1b-842b-75ff6e0bcb1e',
            certificateP12: 'test-cert-p12',
            certificatePassword: 'test-cert-password',
            serialNumber: 'test-serial',
            developerPortalIdentifier: 'test-portal-id',
            validityNotBefore: '2020-10-13T18:28:44.000Z',
            validityNotAfter: '2021-10-13T18:28:44.000Z',
            updatedAt: '2020-10-13T18:38:45.275Z',
            appleTeam: {
              id: '6d39dbfe-a30a-44a4-bc36-d562e5963fa3',
              appleTeamIdentifier: 'test-apple-identifier',
              appleTeamName: null,
              __typename: 'AppleTeam',
            },
            __typename: 'AppleDistributionCertificate',
          },
          provisioningProfile: {
            id: '01517f14-b4cf-489e-b1fd-3f5a61cb8df8',
            expiration: '2021-10-13T18:28:44.000Z',
            developerPortalIdentifier: 'test-profile-id',
            provisioningProfile: 'test-profile',
            updatedAt: '2020-10-29T00:18:32.130Z',
            status: 'active',
            appleDevices: [],
            appleTeam: {
              id: '6d39dbfe-a30a-44a4-bc36-d562e5963fa3',
              appleTeamIdentifier: 'test-apple-identifier',
              appleTeamName: null,
              __typename: 'AppleTeam',
            },
            __typename: 'AppleProvisioningProfile',
          },
          __typename: 'IosAppBuildCredentials',
        },
      ],
      __typename: 'IosAppCredentials',
    };
  }),
};
