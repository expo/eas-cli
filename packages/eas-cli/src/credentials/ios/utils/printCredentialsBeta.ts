import chalk from 'chalk';
import dateformat from 'dateformat';

import {
  CommonIosAppCredentialsFragment,
  IosAppBuildCredentialsFragment,
  IosDistributionType,
} from '../../../graphql/generated';
import log from '../../../log';
import { fromNow } from '../../../utils/date';
import { AppLookupParams } from '../api/GraphqlClient';

export function prettyIosDistributionType(distributionType: IosDistributionType): string {
  switch (distributionType) {
    case IosDistributionType.AppStore:
      return 'App Store';
    case IosDistributionType.AdHoc:
      return 'Ad Hoc';
    case IosDistributionType.Development:
      return 'Development';
    case IosDistributionType.Enterprise:
      return 'Enterprise';
    default:
      return 'Unknown';
  }
}

export function displayEmptyIosCredentials(appLookupParams: AppLookupParams): void {
  const { projectName, bundleIdentifier } = appLookupParams;
  log(chalk.bold(`iOS Credentials`));
  log(`  Project: ${projectName}`);
  log(`  Bundle Identifier: ${bundleIdentifier}`);
  log(`  No credentials set up yet!`);
}

/**
 * sort a build credentials array in descending order of preference
 */
function sortBuildCredentialsByDistributionType(
  iosAppBuildCredentialsArray: IosAppBuildCredentialsFragment[]
): IosAppBuildCredentialsFragment[] {
  // The order in which we choose the distribution type from least to most preferred
  const typePriority = [
    IosDistributionType.Development,
    IosDistributionType.AdHoc,
    IosDistributionType.Enterprise,
    IosDistributionType.AppStore,
  ];
  return iosAppBuildCredentialsArray
    .sort(
      (buildCredentialsA, buildCredentialsB) =>
        typePriority.indexOf(buildCredentialsA.iosDistributionType) -
        typePriority.indexOf(buildCredentialsB.iosDistributionType)
    )
    .reverse();
}

export function displayIosAppCredentials(credentials: CommonIosAppCredentialsFragment): void {
  log(chalk.bold(`iOS Credentials`));
  log(`  Project: ${credentials.app.fullName}`);
  log(`  Bundle Identifier: ${credentials.appleAppIdentifier.bundleIdentifier}`);
  const appleTeam = credentials.appleTeam;
  if (appleTeam) {
    const { appleTeamIdentifier, appleTeamName } = appleTeam;
    log(`  Apple Team: ${appleTeamIdentifier} ${appleTeamName ? `(${appleTeamName})` : ''}`);
  }
  log.newLine();

  if (credentials.iosAppBuildCredentialsArray.length === 0) {
    log(`  Configuration: None setup yet`);
    log.newLine();
    return;
  }
  const sortedIosAppBuildCredentialsArray = sortBuildCredentialsByDistributionType(
    credentials.iosAppBuildCredentialsArray
  );
  for (const iosAppBuildCredentials of sortedIosAppBuildCredentialsArray) {
    displayIosAppBuildCredentials(iosAppBuildCredentials);
  }
}

export function displayIosAppBuildCredentials(
  buildCredentials: IosAppBuildCredentialsFragment
): void {
  log(
    chalk.bold(
      `  Configuration: ${prettyIosDistributionType(buildCredentials.iosDistributionType)}`
    )
  );
  const maybeDistCert = buildCredentials.distributionCertificate;
  log(`  Distribution Certificate:`);
  if (maybeDistCert) {
    const { serialNumber, updatedAt, validityNotAfter, appleTeam } = maybeDistCert;
    log(`    Serial Number: ${serialNumber}`);
    log(`    Expiration Date: ${dateformat(validityNotAfter, 'expiresHeaderFormat')}`);
    if (appleTeam) {
      const { appleTeamIdentifier, appleTeamName } = appleTeam;
      log(`    Apple Team: ${appleTeamIdentifier} ${appleTeamName ? `(${appleTeamName})` : ''}`);
    }
    log(`    Updated ${fromNow(new Date(updatedAt))} ago`);
  } else {
    log(`    None assigned yet`);
  }
  log.newLine();

  const maybeProvProf = buildCredentials.provisioningProfile;
  log(`  Provisioning Profile:`);
  if (maybeProvProf) {
    const { expiration, updatedAt, status, developerPortalIdentifier, appleTeam } = maybeProvProf;
    if (developerPortalIdentifier) {
      log(`    Developer Portal ID: ${developerPortalIdentifier}`);
    }
    log(`    Status: ${status}`);
    log(`    Expiration Date: ${dateformat(expiration, 'expiresHeaderFormat')}`);
    if (appleTeam) {
      const { appleTeamIdentifier, appleTeamName } = appleTeam;
      log(`    Apple Team: ${appleTeamIdentifier} ${appleTeamName ? `(${appleTeamName})` : ''}`);
    }
    log(`    Updated ${fromNow(new Date(updatedAt))} ago`);
  } else {
    log(`    None assigned yet`);
  }
  log.newLine();
}
