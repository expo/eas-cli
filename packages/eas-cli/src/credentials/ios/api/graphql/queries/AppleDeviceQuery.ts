import assert from 'assert';
import { print } from 'graphql';
import gql from 'graphql-tag';

import { ExpoGraphqlClient } from '../../../../../commandUtils/context/contextUtils/createGraphqlClient';
import { DeviceNotFoundError } from '../../../../../devices/utils/errors';
import { withErrorHandlingAsync } from '../../../../../graphql/client';
import {
  AppleDevice,
  AppleDeviceFragment,
  AppleDevicesByAppleTeamQuery,
  AppleDevicesByIdentifierQuery,
  AppleDevicesByTeamIdentifierQuery,
  AppleDevicesByTeamIdentifierQueryVariables,
  AppleTeamFragment,
} from '../../../../../graphql/generated';
import { AppleDeviceFragmentNode } from '../../../../../graphql/types/credentials/AppleDevice';
import { AppleTeamFragmentNode } from '../../../../../graphql/types/credentials/AppleTeam';
import { formatAppleTeam } from '../../../actions/AppleTeamFormatting';

export type AppleDeviceFragmentWithAppleTeam = AppleDeviceFragment & {
  appleTeam: AppleTeamFragment;
};

export type AppleDeviceQueryResult = Pick<
  AppleDevice,
  'id' | 'identifier' | 'name' | 'deviceClass' | 'enabled'
>;

export type AppleDevicesByTeamIdentifierQueryResult = AppleTeamFragment & {
  appleDevices: AppleDeviceQueryResult[];
};

export type AppleDevicesByIdentifierQueryResult = AppleDeviceQueryResult & {
  appleTeam: AppleTeamFragment;
};

export const AppleDeviceQuery = {
  async getAllByAppleTeamIdentifierAsync(
    graphqlClient: ExpoGraphqlClient,
    accountId: string,
    appleTeamIdentifier: string,
    { useCache = true }: { useCache?: boolean } = {}
  ): Promise<AppleDeviceFragmentWithAppleTeam[]> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleDevicesByAppleTeamQuery>(
          gql`
            query AppleDevicesByAppleTeamQuery($accountId: ID!, $appleTeamIdentifier: String!) {
              appleTeam {
                byAppleTeamIdentifier(accountId: $accountId, identifier: $appleTeamIdentifier) {
                  id
                  ...AppleTeamFragment
                  appleDevices {
                    id
                    ...AppleDeviceFragment
                    appleTeam {
                      id
                      ...AppleTeamFragment
                    }
                  }
                }
              }
            }
            ${print(AppleTeamFragmentNode)}
            ${print(AppleDeviceFragmentNode)}
          `,
          {
            accountId,
            appleTeamIdentifier,
          },
          {
            additionalTypenames: ['AppleDevice'],
            requestPolicy: useCache ? 'cache-first' : 'network-only',
          }
        )
        .toPromise()
    );
    assert(
      data.appleTeam.byAppleTeamIdentifier,
      'byAppleTeamIdentifier should be defined in this context - enforced by GraphQL'
    );
    const { appleDevices } = data.appleTeam.byAppleTeamIdentifier;
    assert(appleDevices, 'Apple Devices should be defined in this context - enforced by GraphQL');
    return appleDevices;
  },

  async getAllForAppleTeamAsync(
    graphqlClient: ExpoGraphqlClient,
    { accountName, appleTeamIdentifier, offset, limit }: AppleDevicesByTeamIdentifierQueryVariables
  ): Promise<AppleDeviceFragment[]> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleDevicesByTeamIdentifierQuery>(
          gql`
            query AppleDevicesByTeamIdentifier(
              $accountName: String!
              $appleTeamIdentifier: String!
              $offset: Int
              $limit: Int
            ) {
              account {
                byName(accountName: $accountName) {
                  id
                  appleTeams(appleTeamIdentifier: $appleTeamIdentifier) {
                    id
                    appleTeamIdentifier
                    appleTeamName
                    appleDevices(offset: $offset, limit: $limit) {
                      id
                      identifier
                      name
                      deviceClass
                      enabled
                      model
                    }
                  }
                }
              }
            }
          `,
          { accountName, appleTeamIdentifier, offset, limit },
          {
            additionalTypenames: ['AppleDevice', 'AppleTeam'],
          }
        )
        .toPromise()
    );
    const [appleTeam] = data.account.byName.appleTeams;
    const { appleDevices } = appleTeam;
    if (!appleDevices) {
      throw new Error(`Could not find devices on Apple team -- ${formatAppleTeam(appleTeam)}`);
    }
    return appleDevices;
  },

  async getByDeviceIdentifierAsync(
    graphqlClient: ExpoGraphqlClient,
    accountName: string,
    identifier: string
  ): Promise<AppleDevicesByIdentifierQueryResult> {
    const data = await withErrorHandlingAsync(
      graphqlClient
        .query<AppleDevicesByIdentifierQuery>(
          gql`
            query AppleDevicesByIdentifier($accountName: String!, $identifier: String!) {
              account {
                byName(accountName: $accountName) {
                  id
                  appleDevices(identifier: $identifier) {
                    id
                    model
                    identifier
                    name
                    deviceClass
                    enabled
                    appleTeam {
                      id
                      appleTeamIdentifier
                      appleTeamName
                    }
                  }
                }
              }
            }
          `,
          { accountName, identifier },
          {
            additionalTypenames: ['AppleDevice', 'AppleTeam'],
          }
        )
        .toPromise()
    );

    const device = data.account.byName.appleDevices[0];
    if (!device) {
      throw new DeviceNotFoundError(
        `Device with id ${identifier} was not found on account ${accountName}.`
      );
    }
    return device;
  },
};
