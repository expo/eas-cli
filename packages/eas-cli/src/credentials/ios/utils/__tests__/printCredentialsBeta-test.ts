import mockdate from 'mockdate';
import nullthrows from 'nullthrows';

import Log from '../../../../log';
import { IosAppCredentialsQuery } from '../../api/graphql/queries/IosAppCredentialsQuery';
import { displayIosAppCredentials } from '../printCredentialsBeta';

jest.mock('../../../../log');
jest.mock('chalk', () => ({ bold: jest.fn(log => log) }));

jest.mock('../../api/graphql/queries/IosAppCredentialsQuery');
mockdate.set(new Date('4/20/2021'));

describe('print credentials', () => {
  it('prints the IosAppCredentials fragment', async () => {
    const testIosAppCredentialsData = await IosAppCredentialsQuery.withCommonFieldsByAppIdentifierIdAsync(
      '@foo/bar',
      {
        appleAppIdentifierId: 'test-id',
      }
    );
    displayIosAppCredentials(nullthrows(testIosAppCredentialsData));
    const loggedSoFar = (Log.log as jest.Mock).mock.calls.reduce(
      (acc, mockValue) => acc + mockValue
    );
    expect(loggedSoFar).toMatchSnapshot();
  });
});
