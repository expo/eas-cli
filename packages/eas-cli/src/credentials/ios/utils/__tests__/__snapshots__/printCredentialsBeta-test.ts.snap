// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print credentials prints the IosAppCredentials fragment 1`] = `"[1miOS Credentials[22m  Project: @quinlanj/test52  Bundle Identifier: com.quinlanj.test52  Apple Team: test-apple-identifier [1m  Configuration: App Store[22m  Distribution Certificate:    Serial Number: test-serial    Expiration Date: Wed, 13 Oct 2021 18:28:44 GMT+0000    Apple Team: test-apple-identifier     Updated 6 months ago  Provisioning Profile:    Developer Portal ID: test-profile-id    Status: active    Expiration Date: Wed, 13 Oct 2021 18:28:44 GMT+0000    Apple Team: test-apple-identifier     Updated 5 months ago[1m  Configuration: Ad Hoc[22m  Distribution Certificate:    Serial Number: test-serial    Expiration Date: Wed, 13 Oct 2021 18:28:44 GMT+0000    Apple Team: test-apple-identifier     Updated 6 months ago  Provisioning Profile:    Developer Portal ID: test-profile-id    Status: active    Expiration Date: Wed, 13 Oct 2021 18:28:44 GMT+0000    Apple Team: test-apple-identifier     Updated 5 months ago"`;
